tests/unit/test_charm.py

```diff
@@ -62,12 +62,30 @@ def test_blocked_by_peer_relation_not_ready(self):
         # The BlockStatus is set with a message.
         self.assertEqual(harness.model.unit.status, BlockedStatus("peer relation not ready"))
 
+    def test_blocked_by_missing_num_history_shards(self):
```

Comment:
What about a unit test for when a new config value is added which does not match the previously set value?